src/ImageSharp/Formats/Jpeg/JpegEncoderCore.cs

```diff
@@ -545,8 +571,9 @@ private void WriteStartOfFrame(int width, int height, int componentCount)
             for (int i = 0; i < componentCount; i++)
             {
                 int i3 = 3 * i;
-                this.buffer[i3 + 6] = (byte)(i + 1);
 
+                // Component ID.
+                this.buffer[i3 + 6] = componentIds[i];
```

Comment:
To elide some bound checks slice `buffer` to a Span so that indexing can start at `0` (and write in reverse order, so start at `[2]`, `[1]`, then `[0]`)?

Reply:
changed with 49d86e1b